src/scalems/radical/__init__.py

```diff
@@ -961,14 +976,14 @@ def _connect_rp(execution_manager: 'RPDispatchingExecutor'):
         # Optionally, we could refrain from launching the pilot here, at all,
         # but it seems like a good chance to start bootstrapping the agent environment.
         logger.debug('Launching PilotManager.')
-        pilot_manager = rp.PilotManager(session=execution_manager.session)
-        execution_manager._pilot_manager_uid = pilot_manager.uid
-        logger.debug('Got PilotManager {}.'.format(execution_manager._pilot_manager_uid))
+        pilot_manager = rp.PilotManager(session=runtime.session)
+        logger.debug('Got PilotManager {}.'.format(pilot_manager.uid))
+        runtime.pilot_manager(pilot_manager)
```

Comment:
The current implementation does not capture a reference to the `pilot_manager` in the `runtime` instance, but @andre-merzky says this is wrong. We should explicitly manage the life of the managers through client-held references until `close()` will be called.